calculation/gmhazard_calc/gmhazard_calc/directivity/hypo_sampling.py

```diff
@@ -0,0 +1,256 @@
+        plane_ix = np.flatnonzero(strike_location < plane_lengths)[0]
+        for depth in down_dip_locations:
+            # Copies the planes without any hypocentres and sets the shyp and dhyp values
+            # in the hypocentre plane to the strike location and depth values
+            planes_copy = copy.deepcopy(planes)
+            planes_copy[plane_ix]["shyp"] = strike_location - (total_length / 2)
+            planes_copy[plane_ix]["dhyp"] = depth
+            plane_index.append(plane_ix)
+            planes_list.append(planes_copy)
+
+    return planes_list, plane_index, combo_weights
+
+
+def latin_hypercube_sampling(
+    nhypo: int,
+    planes: Sequence,
+    event_type: EventType,
+    total_length: float,
+    seed: int = None
+):
+    """
+    Using Latin Hypercube to place hypocentres by taking a grid of nhypo length across both strike and dip
+    which is defined by the strike and dip distribution methods but is sectioned to ensure lower probability hypocentre
+    locations are chosen as part of the set
+
+    Parameters
+    ----------
+    nhypo: int
+        Number of hypocentres to use
+    planes: list
+        The planes to adjust and set the hypocentre on
+    event_type: EventType
+        The event type Strike_slip, dip_slip or all for determining the down dip distribution function
+    total_length: float
+        The total length of the fault
+    seed: int
+        The seed to use to get reproducible results
+    """
+    # Define strike and down dip distributions
+    mu, sigma = 0.5, 0.23
+    strike_distribution = stats.truncnorm(
+        (0.0 - mu) / sigma, (1.0 - mu) / sigma, loc=mu, scale=sigma
+    )
+
+    # Works out the depth method for down dip placement of hypocentres
+    # Based on Weilbull or Gamma distributions depending on the EventType
+    if event_type == EventType.DIP_SLIP:
+        down_dip_distribution = GammaTruncated(a=0, b=1)
+    elif event_type == EventType.STRIKE_SLIP:
+        down_dip_distribution = WeibullTruncatedStrikeSlip(a=0, b=1)
+    else:
+        down_dip_distribution = WeibullTruncatedOblique(a=0, b=1)
+
+    # Setup Latin-HyperCube
+    lhd = stats.qmc.LatinHypercube(2, seed=seed)
+    lhd = lhd.random(nhypo)
+    lhd[:, 0] = strike_distribution.ppf(lhd[:, 0])
+    lhd[:, 1] = down_dip_distribution.ppf(lhd[:, 1])
+
+    planes_list = []
+    plane_index = []
+    plane_lengths = np.cumsum(np.asarray([plane["length"] for plane in planes]))
+    # Assuming all widths in the planes are the same
+    assert all([x["width"] == planes[0]["width"] for x in planes])
+
+    # Creates a planes list each with a different hypocentre location based on the strike and depth values
+    # from the latin hypercube distribution
+    for strike, depth in lhd:
+        strike_location = strike * total_length
+        # Get the plane index of the hypocentre location
+        plane_ix = np.flatnonzero(strike_location < plane_lengths)[0]
```

Comment:
They all are actually slightly different in a way which makes this tricky as it is generating a list of different planes per hypocentre location based on the method.
Some require the same strike location per depth and another random of both based on a drawn distribution etc, so I feel a function to do this would just get messy.